lib/vast/Conversion/HLToFunc.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022-present, Trail of Bits, Inc.
+
+#include "vast/Conversion/Passes.hpp"
+#include "vast/Util/Common.hpp"
+
+#include "PassesDetails.hpp"
+
+#include "vast/Conversion/HLToFunc.hpp"
+
+namespace vast {
+
+    using RewritePatternSet = mlir::RewritePatternSet;
+    using FrozenRewritePatternSet = mlir::FrozenRewritePatternSet;
+
+    struct HLToFuncPass : HLToFuncBase< HLToFuncPass >
+    {
+        using HLToFuncBase::HLToFuncBase;
+
+        LogicalResult initialize(MContext *ctx) override {
+            // Build the pattern set within the `initialize` to avoid recompiling PDL
+            // patterns during each `runOnOperation` invocation.
+            RewritePatternSet pattern_list(ctx);
+            populateGeneratedPDLLPatterns(pattern_list);
+            patterns = std::move(pattern_list);
```

Comment:
This is usually done in `runOnOperation` in other C++ only passes - is this the recommended way for this type of pass?

Reply:
you see the comment